src/controllers/controller.h

```diff
@@ -200,6 +206,39 @@ class ControllerJSProxy : public QObject {
         m_pController->send(data, data.length());
     }
 
+    // Available charsets should be available here:
+    // http://www.iana.org/assignments/character-sets/character-sets.xhtml
+    Q_INVOKABLE QJSValue convertEncoding(const QString& targetCharset, QString& value) {
+        std::shared_ptr<QJSEngine> pJsEngine = m_pController->getScriptEngine()->jsEngine();
+        VERIFY_OR_DEBUG_ASSERT(pJsEngine) {
+            return QJSValue::UndefinedValue;
+        }
+
+        icu::UnicodeString unicodeString;
+        UErrorCode errorCode;
+        UConverter* latin9Converter = ucnv_open(targetCharset.toLocal8Bit().data(), &errorCode);
+
+        if (!U_FAILURE(errorCode)) {
+            ucnv_close(latin9Converter);
+            return QJSValue::UndefinedValue;
+        }
+
+        char* result = nullptr;
+        ucnv_fromUChars(latin9Converter,
+                result,
+                0,
+                &value.data()->unicode(),
+                value.length(),
+                &errorCode);
+        ucnv_close(latin9Converter);
```

Comment:
Qt has a wrapper for all this. I think it should be: 
```suggestion
        QTextCodec* codec = QTextCodec::codecForName(targetCharset);
        if (!codec) {
            return QJSValue::UndefinedValue;
        }
        QByteArray result = codec->fromUnicode(value);
```
(Not tested)

Reply:
Unfortunately this nice API was replaced in Qt6 by QStringConverter, which supports much less codecs. The old API is still available in the "Qt 5 Core Compat module" for Qt6, but I guess this is nothing that should be used in new code.

Reply:
Mmm interesting. It would be:

```suggestion
        std::optional<QStringEncoder> encoder = QStringEncoder::forName(targetCharset);
        if (!encoder) {
            return QJSValue::UndefinedValue;
        }
        QByteArray result = encoder->encode(value);
```

Reply:
Our QT vcpkg is build with "icu" 
https://github.com/daschuer/vcpkg/blob/5d58718e04ee9196314dc77d7cf414f36be24b65/ports/qtbase/vcpkg.json#L62

Reply:
@daschuer thank you for pointing this up. If I can't use `QTextCodec`, there's definitely some code I can copy here.

Reply:
I think you can use Qt6`s QStringEncoder

Reply:
Yes, but you need to use Qt 6.6 or higher.

Reply:
QStringEncoder Is available from Qt 6.0, Which part requires Qt 6.6?

Reply:
I looked it up, 6.4 is required, not 6.6. This is the version where support for ICU codecs was added: https://doc.qt.io/qt-6/whatsnew64.html

Reply:
I wonder what the limitations are because from 6.0 the targetCharset name was passed directly as string to ICU. 
Since there is nothing written about the version dependency we may test if it works anyway.

Reply:
Ok, so Qt6.4. Am I guaranteed that Mixxx will be compiled using Qt6.4 or should I add a default behavior when Mixxx is compiled with Qt<6.4.

Reply:
I think our minimum is still Qt 6.2. The proposed code will work with that, but probably with some issue fixed in Qt 6.4 as @JoergAtGithub pointed out. 

My proposal is this: 

If it turns out your original use case works with 6.2, and QStringEncoderQStringEncoder, fine. Only if not, we may consider using ICU dirrectly or the QTextCodec compatibility layer. 

I wonder if QTextCodec than fixes the issue since it is based in the same implementation as QStringEncoder.

Reply:
So if I'm not mistaken, this should look like on my last commit. This, however doesn't work as expected. From JS side, `convertEncoding` returns an object that, printed using `console.log`, display the original string. Not a `TypedArray`. I can't figure out what's happening. I'm leaving it ofr tonight.

Reply:
Maybe do some type introspection (`instanceof`) and if it is an `ArrayBuffer`.

Reply:
Ah you're right. I expected a `TypedArray` and thus a `.forEach()`. So I'd need to do `new Uint8Array(midi.convertEncoding("ISO-8859-15", "Thing to display"))` from JS? Is there a way to directly return a `TypedArray`?

Reply:
You can technically execute some JS code in C++ that essentially does that and return the resulting QJSValue, but thats quite ugly IMO. I think it should up to the caller how they want to interpret that buffer. If thats too much typing for you, you can always create a utility function in your mapping, but lets not bake that into the API. It will cause API inconsistencies and unnecessary overhead.